### PR TITLE
Feat: Run main.py to debug python back end

### PR DESCRIPTION
**Pull request by joelsprunger1**
# Description
This adds an
`if __name__ == "__main__":`

statement to main.py as well as using dotenv to create environment variables in main.py for the purposes of debugging the backend code. 

Now main.py can be run in an IDE to start the ASGI backend and the docker-compose-front.yml can be used to serve the front end. 

## Checklist before requesting a review

Please delete options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented hard-to-understand areas
- [x] I have ideally added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

### CODE REVIEW

File: backend/core/main.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import os
+from dotenv import load_dotenv
+load_dotenv()
```
Comment (StanGirard) [inline backend/core/main.py:3]: We already use Loadenv kind of with BaseSettings from Pydantic


https://docs.pydantic.dev/latest/usage/pydantic_settings/

File: docker-compose-front.yml
Comment (StanGirard) [inline docker-compose-front.yml:1]: This file isn't required. you can target with `docker compose up frontend`

**Review by StanGirard — CHANGES_REQUESTED** [review]

**Comment by StanGirard** [comment]
Thanks a lot for the PR. I don't see the issue you are trying to tackle :/ It should be working already.

**Comment by joelsprunger1** [comment]
This allows anyone to open the project in the IDE, put a breakpoint in the backend somewhere and run main.py. If there is a better/recommended way to do this, like running a different script, or a adding a command in the IDE please let me know.

File: backend/core/main.py (hunk 1 above)
Comment (StanGirard) [reply backend/core/main.py:3]: Can you put that in the 

if __name__ == "__main__":

So it doesn't disrupt anything just in case ;)

Comment (joelsprunger1) [reply backend/core/main.py:3]: Yes - I still need it at the top of the file - but I will also wrap it in name==main so that it doesn't execute otherwise

**Comment by joelsprunger1** [comment]
Okay @StanGirard this should be good to merge.

**Review by joelsprunger1 — COMMENTED** [review]
This should be good.

**Review by StanGirard — APPROVED** [review]

**Comment by StanGirard** [comment]
There are conflicts :/

**Comment by gozineb** [comment]
Hey @joelsprunger1 ! Thanks for this PR ! Have you been able to fix the conflicts so it can be merged ?

**Comment by joelsprunger** [comment]
This can be closed.

**Pull request closed without merging**